src/Runtime/LoopIterator.php

```diff
@@ -63,8 +74,13 @@ public function next(): void
     public function rewind(): void
     {
         $this->seq->rewind();
-        $this->previous = ['valid' => false, 'key' => null, 'value' => null];
-        $this->current = ['valid' => $this->seq->valid(), 'key' => $this->seq->key(), 'value' => $this->seq->current()];
+        if ($this->seq->valid()) {
+            $this->previous = ['valid' => false, 'key' => null, 'value' => null];
```

Comment:
shouldn't we always assign `previous` ?

Reply:
That was something i wonder, but there is only one situation getPrevious() can be called without beeing set, it's if someone call "loop.previous" in the `else` part of its `for`  ... `else` .... `endfor`.

Because the node inside the `for` block can only be called if there is _something_ in the loop (so when this->seq->isValid() is true)

Reply:
and in such case, we still want it to be able to return `null` as value, not a PHP error

Reply:
See #4169